federation/pkg/federation-controller/ingress/ingress_controller.go

```diff
@@ -53,6 +53,11 @@ const (
 	uidConfigMapName        = "ingress-uid"                                 // Name of the config-map and key the ingress controller stores its uid in.
 	uidConfigMapNamespace   = "kube-system"
 	uidKey                  = "uid"
+	// Annotation on the ingress in federation control plane that is used to keep
+	// track of the first cluster in which we create ingress.
+	// We wait for ingress to be created in this cluster before creating it any
+	// other cluster.
+	firstClusterAnnotation = "ingress.federation.kubernetes.io/first-cluster"
```

Comment:
I think "master" is an overloaded term that can be interpreted as a lot of things.
This is really just a "first" cluster where we create ingress.
That being said, I was myself on the fence between master and first. I went with first due to the reason above.

So I prefer first but dont mind changing it to master if you have a strong opinion.

Re: the key, why do you think it should be federation.ingress.kubernetes.io? I didnt find an example of us using that pattern anywhere. I thought of it as subdomains. federation is a subdomain of kubernetes and ingress is a further subdomain of federation. But can change it if we are using the other patter somewhere already?
Another options is `federation.kubernetes.io/ingress-first-cluster` :)